packages/compass-data-modeling/src/components/diagram-editor.tsx

```diff
@@ -335,9 +345,20 @@ const DiagramEditor: React.FunctionComponent<{
             title={diagramLabel}
             edges={edges}
             nodes={areNodesReady ? nodes : []}
-            onEdgeClick={() => {
-              // TODO: we have to open a side panel with edge details
-              onOpenSidePanel();
+            // With threshold too low clicking sometimes gets confused with
+            // dragging
+            // @ts-expect-error expose this prop from the component
+            nodeDragThreshold={3}
+            // @ts-expect-error expose this prop from the component
+            onNodeClick={(_evt, node) => {
+              if (node.type !== 'collection') {
+                return;
+              }
+              onCollectionSelect(node.id);
+            }}
```

Comment:
Note to self: don't forget to open tickets for those changes in diagramming package